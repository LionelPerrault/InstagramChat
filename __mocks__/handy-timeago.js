export default () => 'a month ago'
