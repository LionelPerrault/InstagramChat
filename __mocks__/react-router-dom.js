export const Link = 'Link'
export const Switch = 'Switch'
export const Redirect = 'Redirect'
export const Route = 'Route'
export const NavLink = 'NavLink'
