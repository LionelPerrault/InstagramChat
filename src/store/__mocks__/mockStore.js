import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
