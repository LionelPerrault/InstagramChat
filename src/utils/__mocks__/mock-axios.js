import AxiosAdaper from 'axios-mock-adapter'
import axios from 'axios'
const mockAxios = new AxiosAdaper(axios)

/**
 * Mocks Axios requests.
 *
 * @param {String} endpoint URL to request data from.
 * @param {any} data Fake data to return.
 */
export const mockAxiosRequest = (endpoint, data) =>
  mockAxios.onPost(`/api/${endpoint}`).reply(200, data)
