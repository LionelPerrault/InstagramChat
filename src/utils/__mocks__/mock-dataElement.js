/**
 * Creates a fake div.data with required data attributes for use in tests.
 * @returns {HTMLDivElement} dataElementDIV
 */
const MockDataElement = () => {
  let dataElement = document.createElement('div')
  dataElement.setAttribute('class', 'data')
  dataElement.setAttribute('data-session', '24')
  dataElement.setAttribute('data-username', 'takkar')
  dataElement.setAttribute('data-email-verified', 'no')
  dataElement.setAttribute('data-isadmin', 'false')
  document.body.prepend(dataElement)

  return dataElement
}

export default MockDataElement
