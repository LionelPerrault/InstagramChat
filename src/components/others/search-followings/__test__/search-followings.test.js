import React from 'react'
import { create } from 'react-test-renderer'
import SearchFollowings from '../search-followings'
import { shallow, mount } from 'enzyme'
import { mockAxiosRequest } from '../../../../utils/__mocks__/mock-axios'
import usersMockArray from './users-mockArray'

describe('Search-Followings Component', () => {
  const mockFn = jest.fn()
  const comp = <SearchFollowings when="new_con" done={mockFn} />

  mockAxiosRequest('search-followings', usersMockArray)

  it('should match snapshot', () => {
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should get users from API on componentDidMount', async () => {
    const wrapper = shallow(comp)
    await wrapper.instance().componentDidMount()
    expect(wrapper.state().data).toBeArray()
    expect(wrapper.state().data[0]).toContainKeys([
      'follow_to',
      'follow_to_username',
    ])
  })

  const simulateChange = (wrapper, value) => {
    wrapper.find('input').simulate('change', { target: { value } })
  }

  it('should return matched users when typing', async () => {
    const wrapper = mount(comp)
    await wrapper.instance().componentDidMount()

    // should return coldplay as 'cold' matches only coldplay
    simulateChange(wrapper, 'cold')
    expect(wrapper.state().followings).toBeArrayOfSize(1)
    expect(wrapper.state().followings[0]).toContainEntries([
      ['follow_to', 8],
      ['follow_to_username', 'coldplay'],
    ])

    // should return ghalib as 'gh' matches ghalib only
    simulateChange(wrapper, 'gh')
    expect(wrapper.state().followings).toBeArrayOfSize(1)
    expect(wrapper.state().followings[0]).toContainEntries([
      ['follow_to', 7],
      ['follow_to_username', 'ghalib'],
    ])

    // should return [] as 'very-diff' matches neither ghalib nor coldplay
    simulateChange(wrapper, 'very-diff')
    expect(wrapper.state().followings).toBeArrayOfSize(0)
  })
})
