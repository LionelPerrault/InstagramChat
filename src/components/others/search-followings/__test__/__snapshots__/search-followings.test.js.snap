// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search-Followings Component should match snapshot 1`] = `
<div
  className="search_followings"
>
  <div
    className="p_add_taggings"
  >
    <input
      autoComplete="false"
      className="p_add_taggings_text"
      disabled={false}
      maxLength="255"
      onChange={[Function]}
      placeholder=""
      spellCheck="false"
      type="text"
      value=""
    />
  </div>
</div>
`;
